src/core/server/legacy/config/legacy_object_to_config_adapter.ts

```diff
@@ -33,18 +34,24 @@ interface LegacyLoggingConfig {
   events?: Record<string, string>;
 }
 
+type MixedLoggingConfig = LegacyLoggingConfig & Partial<LoggingConfigType>;
+
 /**
  * Represents adapter between config provided by legacy platform and `Config`
  * supported by the current platform.
  * @internal
  */
 export class LegacyObjectToConfigAdapter extends ObjectToConfigAdapter {
-  private static transformLogging(configValue: LegacyLoggingConfig = {}) {
+  private static transformLogging(configValue: MixedLoggingConfig = {}) {
+    const { appenders, root, loggers, ...legacyLoggingConfig } = configValue;
+
     const loggingConfig = {
       appenders: {
-        default: { kind: 'legacy-appender', legacyLoggingConfig: configValue },
+        ...appenders,
+        default: { kind: 'legacy-appender', legacyLoggingConfig },
       },
-      root: { level: 'info' },
+      root: { level: 'info', ...root },
```

Comment:
Note: We allow users to configure roots. Although if a user specifies `root.appenders` without `defaults` NP logging service won't send logs to the legacy system.

Reply:
Logging a warning makes sense to me 👍

Reply:
added config validation to prevent server start